src/app/examples/grid-angular.component.ts

```diff
@@ -349,40 +353,24 @@ export class GridAngularComponent implements OnInit {
   }
 
   /* Create an Action Dropdown Menu */
+
   deleteCell(rowNumber: number) {
     const item = this.angularGrid.dataView.getItem(rowNumber);
     this.angularGrid.gridService.deleteItemById(item.id);
   }
 
   onActiveCellChanged(event, args) {
-    if (args.cell !== 6) {
-      return; // don't do anything unless it's the Action column which is at position 6 in this grid
-    }
-
-    $('#myDrop').remove(); // make sure to remove previous Action dropdown, you don't want to have 100 after a 100 clicks...
-    const cell = args.cell;
-    const row = args.row;
-
-    // hide the dropdown we created as a Formatter, we'll redisplay it later
-    const cellPos = $(`#myDrop-r${row}-c${cell}`).offset();
-
-    const componentOutput = this.angularUtilService.createAngularComponent(CustomActionFormatterComponent);
-
-    // pass "this" and the row number to the Component instance (CustomActionFormatter) so that we can call "parent.deleteCell(row)" with (click)
-    Object.assign(componentOutput.componentRef.instance, { parent: this, row: args.row });
-
-    // use a delay to make sure Angular ran at least a full cycle and make sure it finished rendering the Component before using it
-    setTimeout(() => {
-      const elm = $(componentOutput.domElement);
-      elm.appendTo('body');
-      elm.css('position', 'absolute');
-      elm.css('top', cellPos.top + 5);
-      elm.css('left', cellPos.left);
-      $('#myDrop').addClass('open');
-
-      $('#myDrop').on('hidden.bs.dropdown', () => {
-        $(`#myDrop-r${row}-c${cell}`).show();
-      });
-    });
+    DropDownService.init(
+      {
+        angularUtilService: this.angularUtilService,
+        angularGrid: this.angularGrid,
+        CustomActionFormatter: CustomActionFormatterComponent,
+        args,
+        actionColumnIndex: 6, 
+        parent : this,
+        $
```

Comment:
there's no need to pass the `$`, you can use `declare var $: any;` in the Service.

Reply:
ok